### DISCUSSION TITLE
SAML 2.0 Failing

### DISCUSSION THREAD

**Discussion by cmonty** [structurizr/onpremises#27] [Q&A] [answered]
Docker image (tag: `2996`) throws an unsupported method error for `/saml/SSO` after redirecting to IdP.

```
[WARN ] 2023-03-01 21:44:47.408 [http-nio-8080-exec-5] DefaultHandlerExceptionResolver - Resolved [org.springframework.web.HttpRequestMethodNotSupportedException: Request method 'POST' not supported]
```

Is POST not supported? Did I configure something incorrectly?

**Comment by cmonty** [comment]
@simonbrowndotje do you have any thoughts on why Structurizr is throwing unsupported method here?

**Reply by simonbrowndotje** [reply]
You haven't really provided any information I'm afraid, so it's impossible to say what's happening. My guess would be that you've configured something incorrectly.

**Reply by cmonty** [reply]
@simonbrowndotje is there a way to configure what SSL certificate is used for signing using the Docker image? My IdP requires signing of the request and the certificate used by default is a Structurizr certificate that expired in 2019.

**Reply by simonbrowndotje** [reply, marked as answer]
Not with the pre-built Docker image, you'd need to provide your own certificate by building from source or unpacking the .war file version; see https://docs.spring.io/spring-security-saml/docs/current/reference/htmlsingle/#security in the Spring SAML docs.
